### ISSUE TITLE
Support GitHub action tokens in validation regex

### ISSUE THREAD

**Issue by Beanow** [sourcecred/sourcecred#1461]
Testing GitHub actions to run SourceCred appear to have a different format for the tokens.

This regex considers the token to be invalid, while bypassing this check the token can be used for the GraphQL queries without any problems.
https://github.com/sourcecred/sourcecred/blob/51e3eb8c2557185da3e1536229f228b29ddebc7d/src/plugins/github/fetchGithubRepo.js#L40-L43

It would make sense to me to obtain an example of what these action tokens look like to patch the regex to support them.

Example failing run:
https://github.com/sourcecred/sourcecred-action/runs/301714169
```
SOURCECRED_GITHUB_TOKEN=${TOKEN} docker run -i -v sourcecred_data:/data \
  -e SOURCECRED_GITHUB_TOKEN sourcecred/sourcecred load ${GITHUB_REPOSITORY}
...
  GO   load-sourcecred/sourcecred-action
  GO   github/sourcecred/sourcecred-action
Error: Invalid token: ***
    at fetchGithubRepo (webpack:///./src/plugins/github/fetchGithubRepo.js?:33:158)
    at loadGraph (webpack:///./src/plugins/github/loadGraph.js?:17:283)
    at load (webpack:///./src/api/load.js?:26:173)
    at async loadCommand (webpack:///./src/cli/load.js?:14:33)
```

Example of running with https://github.com/sourcecred/sourcecred/commit/b425b009cbbeb9caa26225ed59dc1c2d37c80847 patch:
https://github.com/sourcecred/sourcecred-action/runs/310910767
```
SOURCECRED_GITHUB_TOKEN=${TOKEN} docker run -i -v sourcecred_data:/data \
  -e SOURCECRED_GITHUB_TOKEN sourcecred/sourcecred:ignore-token-regex \
  load ${GITHUB_REPOSITORY}
...
  GO   load-sourcecred/sourcecred-action
  GO   github/sourcecred/sourcecred-action
Looks like an invalid token: ***
 DONE  github/sourcecred/sourcecred-action: 873ms
  GO   compute-cred
 DONE  compute-cred: 9ms
 DONE  load-sourcecred/sourcecred-action: 901ms
```

cc @vsoch @wchargin

**Comment by vsoch** [comment]
hey! Here is feedback about formatting from Francis

> Thanks for writing back! What you're asking is within scope of what we can help with and I can help you by providing in this thread. I politely decline to comment in the public issue, though I appreciate you asking.
> Because the GITHUB_TOKEN secret is a GitHub App installation access token:
> https://help.github.com/en/actions/automating-your-workflow-with-github-actions/authenticating-with-the-github_token#about-the-github_token-secret
> You can rely on the publicly documented example of how an installation access token is formatted:
> https://developer.github.com/apps/building-github-apps/authenticating-with-github-apps/#authenticating-as-an-installation
> Our public documentation guarantees expected, stable behavior. I hope this helps you with your project!

**Comment by Beanow** [comment]
Guessing this is the example one: `"token": "v1.1f699f1069f60xxx"`?
So a `v1.` prefix and seemingly hexadecimal from there, at least 16 characters.
Probably still good to collect a few real world examples to see if those 16 characters hex match up or they're actually longer.

**Comment by vsoch** [comment]
If we start with that format, we could run the nightly job (and would be alerted if there are any mismatches. Of course we wouldn't have the token in question, but it would prompt us to look again!

**Comment by Beanow** [comment]
Well I'm sure we could trick the output log masking and uncover one such token. My concern there would be revoking the token for security. I haven't found a revoke option, but apparently they expire in 1 hour.

https://developer.github.com/v3/apps/#create-a-new-installation-token
> Creates an installation access token that enables a GitHub App to make authenticated API requests for the app's installation on an organization or individual account. Installation tokens expire one hour from the time you create them. Using an expired token produces a status code of 401 - Unauthorized, and requires creating a new installation token.

Alternatively we could manually create a GitHub app and request such a token for inspection. It seems to be a somewhat involved process with managing of a keypair and generating JWTs. But doable.

But in the end, I think this validation step's primary utility is to help users do the right thing. A good error message can help them fix problems when providing a bad token. Having just a warning message instead of a crash and letting the GitHub API decide whether it's actually invalid seems like a fine approach to me as well.

I can make a guesstimate new regex, but keep it as a warning only.
What do you think @decentralion?

**Comment by decentralion** [comment]
Your approach seems fine to me @Beanow

**Comment by Beanow** [comment]
Hey @vsoch #1471 was merged. If you could test with the latest `:dev` build of the image, that would be great :]

**Comment by vsoch** [comment]
Testing now!

**Comment by Beanow** [comment]
`Warning: GitHub installation access token has an unexpected hexadecimal component length of 40.`
As expected it wasn't 16, like the docs were showing :sweat_smile:.
But it does detect it's like an installation token and just warns about this length.

**Comment by vsoch** [comment]
Woo! Worked like a charm! And so did you relative link of the svg.

The contributors.svg showed in [this readme](https://github.com/sourcecred/sourcecred-action) (super boring, but it's ok :)) was generated by the action, which did:

 1. generation with sourcecred/sourcecred:dev
 2. open pull request to update the graphic

and then I merged to master!

I set it as a scheduled task to run weekly - so I'm going to wait for one clean run (if you want to contribute between now and then it would be nice to have another picture on there!) and then I'll write it up and advertise a bit for people to (unofficially) use. Copy pasting these files into your .github/workflows folder and adding a link in some README is a really easy way to use sourcecred without needing to worry about installing dependencies, containers, etc.
